### ISSUE TITLE
Remove GTK 1.0 utilities from Twisted

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3699]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#3699|
|Type|enhancement|
|Created|2009-03-23 03:00:16Z|
|Branch|https://github.com/twisted/twisted/tree/nogtk1-3699|

As far as I know, nobody is using these.  They are unmaintained.  I don't know if they work.  While it's technically possible to run them on a supported Ubuntu desktop, you need to be using Dapper to do so and you need to have packages from universe installed to do it.

I'd like to encourage better support and more development of the more recent, supported, GTK2 utilities, but I feel like the weight of the collected unmaintained code is as much a disincentive to do that as the lack of time.

So let's get rid of them.

<details><summary>Searchable metadata</summary>

```
trac-id__3699 3699
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_nogtk1_3699 branches/nogtk1-3699
branch_author__glyph glyph
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1237777216000000 1237777216000000
changetime__1241305299000000 1241305299000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph commented|
|-|-|

(In [26518]) Branching to 'nogtk1-3699'

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

Note that I don't include the reactor (hence "utilities"), since:

  1. unlike any of the other code, it's actually tested (albeit indirectly) by the reactor tests and the 'reactors' buildslave, and
  2. it's still possible that someone might need to write a gtk1 app for some pathological reason, but I'm 99% sure there's no reason that anyone would need to use any of that other code.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph removed owner|
|-|-|

That should be just about it.  If anyone wants to raise a policy objection we can always add deprecations instead, but I think it would be a waste of effort.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph commented|
|-|-|

Builds kicked off, test results should be here: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/nogtk1-3699

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

This is related to [[#2833](https://github.com/twisted/twisted/issues/2833)](https://github.com/twisted/twisted/issues/2833) and [[#3340](https://github.com/twisted/twisted/issues/3340)](https://github.com/twisted/twisted/issues/3340).  I think it'd be good to take care of #2833 soon, even if we leave gtkreactor in for a while longer.

This branch also half resolves #3340 as is - it deletes most of the implementation of `im`, but it doesn't delete `bin/words/im` or `twisted/words/scripts/im.py`.  It probably should, and then it will completely resolve [#3340](https://github.com/twisted/twisted/issues/3340).

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_6"></a>|@mwhudson set owner to @glyph|
|-|-|

twisted/scripts/manhole.py appears to import stuff removed by this branch.

grepping for gtk finds another bunch of crap in manhole that I can't really tell if it should be deleted or not.  Maybe you can :)

The rest looks fine.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph removed owner|
|-|-|

I've also deleted some old Jython junk which doesn't work any more either in this branch.

I've removed bin/words and twisted/words/scripts, so this should resolve [#3340](https://github.com/twisted/twisted/issues/3340) as well.

mwh, thanks for drawing my attention to `spelunk_gnome` - it's also gone now, which should be the last of the gtk 1.x stuff in Twisted.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @glyph|
|-|-|

Please update the copyright on twisted/scripts/manhole.py.  The rest seems good, please merge (and go ahead and resolve [#3340](https://github.com/twisted/twisted/issues/3340) as well).

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_9"></a>|@glyph commented|
|-|-|

(In [26800]) re [#3699](https://github.com/twisted/twisted/issues/3699) Update copyright statement as per review.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_10"></a>|@glyph set status to `closed`|
|-|-|

(In [26801]) Remove a bunch of grotty old non-working GTK1 and Swing code.

This eliminates all of the gtk1 code remaining in Twisted besides the explicit
gtk1 reactor support.

Author: glyph

Reviewer: mwh, exarkun

Fixes [#3699](https://github.com/twisted/twisted/issues/3699)

Fixes [#3340](https://github.com/twisted/twisted/issues/3340)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
